tests/test_twitter_ads_tweets_bookmark.py

```diff
@@ -0,0 +1,205 @@
+
+                # Verify the second sync sets both bookmarks of the expected form
+                self.assertIsNotNone(first_scheduled_bookmark_value_utc)
+                self.assertIsNotNone(second_scheduled_bookmark_value_utc)
+
+                # Verify the second sync bookmark is Equal to the first sync bookmark
+                # assumes no changes to data during test
+                self.assertEqual(second_published_bookmark_value_utc, first_published_bookmark_value_utc)
+                self.assertEqual(second_scheduled_bookmark_value_utc, first_scheduled_bookmark_value_utc)
+
+                for record in first_sync_messages:
+
+                    # Verify the first sync bookmark value is the max replication key value for a given stream
+                    replication_key_value = record.get(replication_key)
+                    if record.get('tweet_type') == "PUBLISHED":
+                        # Verify the first sync bookmark value is the max replication key value for a PUBLISHED tweets
+                        self.assertLessEqual(
+                            replication_key_value, first_published_bookmark_value_utc,
+                            msg="First sync bookmark for PUBLISHED tweets was set incorrectly, a record with a greater replication-key value was synced."
+                        )
+                    else:
+                        # Verify the first sync bookmark value is the max replication key value for a SCHEDULED tweets
+                        self.assertLessEqual(
+                            replication_key_value, first_scheduled_bookmark_value_utc,
+                            msg="First sync bookmark for SCHEDULED tweets was set incorrectly, a record with a greater replication-key value was synced."
+                        )
+                
+                for record in second_sync_messages:
+                    # Verify the second sync replication key value is Greater or Equal to the first sync bookmark
+                    replication_key_value = record.get(replication_key)
+
+
+                    # Verify the second sync bookmark value is the max replication key value for a given stream
+                    if record.get('tweet_type') == "PUBLISHED":
+                        # Verify the second sync replication key value is Greater or Equal to the first sync bookmark
+
+                        self.assertGreaterEqual(replication_key_value, simulated_published_bookmark_value,
+                                                msg="Second sync records do not repect the previous bookmark for PUBLISHED tweets.")
+
+                        self.assertLessEqual(
+                            replication_key_value, second_published_bookmark_value_utc,
+                            msg="Second sync bookmark for PUBLISHED tweets was set incorrectly, a record with a greater replication-key value was synced."
+                        )
```

Comment:
If the expectation is that the tap will use the start date if no `SCHEDULED` tweets bookmark is saved, what is the purpose of saving a bookmark based on the `PUBLISHED` tweets? Will this ever be used by the tap?

Reply:
The tap makes two separate calls to fetch records of `PUBLISHED` and `SCHEDULED` tweets.  Bookmark based on `PUBLISHED` tweets would be used to fetch records of `PUBLISHED` tweets only. Tap would use the start date to fetch `SCHEDULED` tweets if a bookmark for SCHEDULED tweets is not available in the state.